### ISSUE TITLE
[ui5-toolbar]: overflow menu missing list read out

### ISSUE THREAD

**Issue by ivoplashkov** [SAP/ui5-webcomponents#12605]
### Bug Description

The dropdown menu triggered by the “Additional Options” (toolbar overflow) button is currently announced by screen readers as an “Available Values dialog.” This creates confusion for non-sighted users because:

The control is first announced as a menu button but then as a dialog, which is inconsistent.

The menu structure is not conveyed properly to assistive technologies.

Menu items are not announced as part of a list, and their positions (e.g., “1 of 9,” “2 of 9”) are not read out.

As a result, screen reader users are unable to understand that they are navigating within a list of menu items or how many options are available.

### Affected Component

ui5-toolbar

### Expected Behaviour

The component should be announced as a menu, not as a dialog.

Each menu item should be announced as part of a structured list, with position feedback (e.g., “1 of 9,” “2 of 9,” etc.) to clearly communicate the number and order of options.

Ensure ARIA roles and attributes (e.g., role="menu", role="menuitem", aria-setsize, aria-posinset) are correctly implemented.

<img width="1920" height="1060" alt="Image" src="https://github.com/user-attachments/assets/d7bde0d9-e4be-4bfe-a65c-ce0e42d6394f" />

### Isolated Example

https://ui5.github.io/webcomponents/nightly/components/Toolbar/#always-overflowing-items

### Priority

Medium

### UI5 Web Components Version

2.16

### Browser

Chrome

### Additional Context

This issue was reported in the Rich Text Editor component, which internally uses the ui5-toolbar - https://github.tools.sap/sap-webcomponents/rich-text-editor/issues/424.

### Organization

SAP Successfactors

### Declaration

- [x] I’m not disclosing any internal or sensitive information.
